src/cdk/overlay/overlay.ts

```diff
@@ -111,7 +112,12 @@ export class Overlay {
    * @returns A portal outlet for the given DOM element.
    */
   private _createPortalOutlet(pane: HTMLElement): DomPortalOutlet {
+    // We have to resolve the ApplicationRef later in order to allow people
+    // to use overlay-based providers during app initialization.
+    if (!this._appRef) {
```

Comment:
This has been like this for a long time- did something change in Angular 6.0?

Reply:
I don't think anything has changed, but it's just not that common of a use case. FWIW, we had something similar for dialog, but I had a hard time reproducing it in the demo app so I never got around to fixing it https://github.com/angular/material2/issues/10304.